internal/pkg/aws/elbv2/elbv2.go

```diff
@@ -33,6 +35,44 @@ func New(sess *session.Session) *ELBV2 {
 	}
 }
 
+// ListenerRuleHostHeaders returns all the host headers for a listener rule.
+func (e *ELBV2) ListenerRuleHostHeaders(ruleARN string) ([]string, error) {
+	resp, err := e.client.DescribeRules(&elbv2.DescribeRulesInput{
+		RuleArns: aws.StringSlice([]string{ruleARN}),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("get listener rule for %s: %w", ruleARN, err)
+	}
+	if len(resp.Rules) == 0 {
+		// Ideally this should never happen.
+		return nil, fmt.Errorf("cannot find listener rule %s", ruleARN)
+	}
+	rule := resp.Rules[0]
+	hostHeaderSet := make(map[string]bool)
+	for _, condition := range rule.Conditions {
+		if aws.StringValue(condition.Field) == "host-header" {
+			for _, value := range condition.Values {
+				hostHeaderSet[aws.StringValue(value)] = true
+			}
+			if condition.HostHeaderConfig == nil {
+				// Ideally this should never happen.
+				continue
+			}
+			for _, value := range condition.HostHeaderConfig.Values {
+				hostHeaderSet[aws.StringValue(value)] = true
+			}
+			// Each rule can optionally include up to one of each of the conditions.
+			break
```

Comment:
can you help me understand why we `break` here, but `continue` on L59?

Reply:
Good call! L59 we should break as well